presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -353,6 +369,8 @@ public void addParametricType(ParametricType parametricType)
                         return Optional.of(getType(new TypeSignature(resultTypeBase)));
                     case StandardTypes.VARCHAR:
                         return Optional.of(createVarcharType(0));
+                    case StandardTypes.CHAR:
+                        return Optional.of(createCharType(1));
```

Comment:
Why invalid? The spec says "The value of a <length> shall be greater than 0 (zero).", but if we're following the spec then `varchar(0)` should be invalid too. I think it would be better to make it consistent and implement `char(0)` unless that causes an obvious problem